src/WebJobs.Extensions.DurableTask/ProviderUtils.cs

```diff
@@ -16,6 +19,22 @@ namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
     /// </summary>
     public static class ProviderUtils
     {
+#if !FUNCTIONS_V1
+        /// <summary>
+        /// Providers can implement this interface to override the autoscaling.
+        /// </summary>
+        public interface IProviderWithAutoScaling
```

Comment:
`DurabilityProvider` is itself a class with virtual methods, so I think we can substantially simplify the design by just making `TryGetScaleMonitor` a virtual method that by default returns `PsuedoScaleMonitor`, and `DurabilityProvider` implementations that want to support runtime driven scale just override that method. That way we don't need to worry about this optional interface on the implementations or when retrieving scale monitors.

Reply:
I agree that it makes more sense. The problem was that in order to treat AzureStorage the same as other providers, I have to pass a whole lot of parameters into the call that seem superfluous (e.g. taskhub name, connections string) because the provider should already know these (but AzureStorageDurabilityProvider does not). I wanted to change existing code as little as possible.

Reply:
If you have some spare cycles, feel free to clean this up.